ssg/playbook_builder.py

```diff
@@ -0,0 +1,275 @@
+
+    def get_profile_selections(self, profile):
+        """
+        Provides a tuple (rules, variables) where rules is a list of rules
+        selected in the profile and variables is a dictionary of
+        variables and their values in the profile. The method can handle
+        profiles which extend other profiles.
+        """
+        rules = []
+        variables = dict()
+        if profile.extends:
+            extended_profile_path = os.path.join(
+                self.profiles_dir, profile.extends + ".profile")
+            extended_profile = ssg.build_yaml.Profile.from_yaml(
+                extended_profile_path)
+            if not profile:
+                sys.stderr.write(
+                    "Could not parse profile %s.\n" % extended_profile_path)
+                return None
+            rules, variables = self.get_profile_selections(extended_profile)
+        rules.extend(profile.get_rule_selectors())
+        variables.update(profile.get_variable_selectors())
+        return rules, variables
+
+    def choose_variable_value(self, var_id, variables, refinements):
+        """
+        Determine value of variable based on profile refinements.
+        """
+        if var_id in refinements:
+            selector = refinements[var_id]
+        else:
+            selector = "default"
+        try:
+            options = variables[var_id]
+        except KeyError:
+            raise ValueError("Variable '%s' doesn't exist." % var_id)
+        try:
+            value = options[selector]
+        except KeyError:
+            if len(options.keys()) == 1:
+                # We will assume that if there is just one option that it could
+                # be selected as a default option.
+                value = list(options.values())[0]
+            else:
+                raise ValueError(
+                    "Selector '%s' doesn't exist in variable '%s'. "
+                    "Available selectors: %s." %
+                    (selector, var_id, ", ".join(options.keys()))
+                )
+        return value
+
+    def get_data_from_snippet(self, snippet_yaml, variables, refinements):
+        """
+        Extracts and resolves tasks and variables from Ansible snippet.
+        """
+        xccdf_var_pattern = re.compile(r"\(xccdf-var\s+(\S+)\)")
+        tasks = []
+        values = dict()
+        for item in snippet_yaml:
+            if isinstance(item, str):
+                match = xccdf_var_pattern.match(item)
+                if match:
+                    var_id = match.group(1)
+                    value = self.choose_variable_value(var_id, variables,
+                                                       refinements)
+                    values[var_id] = value
+                else:
+                    raise ValueError("Found unknown item '%s'" % item)
+            else:
+                tasks.append(item)
+        return tasks, values
+
+    def get_tags_from_comments(self, snippet_path):
```

Comment:
Are we sure that we need this? Enhanced snippets should already contain those.

Reply:
Ah, I forgot to remove it, it's not used anywhere.

Reply:
I have removed unused method get_tags_from_comments